src/pages/SwapNew/components/SwapInfo/SwapInfoDetailed.tsx

```diff
@@ -0,0 +1,72 @@
+import { Trade, TradeType } from '@swapr/sdk'
+
+import { motion } from 'framer-motion'
+import { useCallback, useContext } from 'react'
+import styled from 'styled-components'
+
+import { SwapContext } from '../../../Swap/SwapBox/SwapContext'
+import { SwapDexInfoItem } from './SwapDexInfoItem'
+
+type SwapInfoDetailedProps = {
+  loading: boolean
+  allPlatformTrades?: (Trade | undefined)[]
```

Comment:
No need for undefined type in `allPlatformTrades`, as it is already being checked in the parent component.

Reply:
There actually is, because for the platform whose trade details can not be fetched return type is `undefined`.